### ISSUE TITLE
Emacs Failed to Build on OS X Yosemite v10.10.5 With Options Specified in Issue Body

### ISSUE THREAD

**Issue by RandomDSdevel** [Homebrew/legacy-homebrew#43619]
Options used:  
```
--with-cocoa
--with-d-bus
--with-glib
--with-gnutls
--with-imagemagick
--with-librsvg
--with-mailutils
--with-x11
```

[Here](https://github.com/RandomDSdevel/Homebrew-Installation-Logs/tree/master/Emacs/Trial%201) are the related logs.  

Possible related issues:  

* #37803 ('emacs can be linked against XQuartz or Brew font-config')
* #36751 ('Emacs 24.4.1:  ns-antialias-text still unrecognized')

**Comment by dunn** [comment]
That's interesting:
```
checking for library containing tputs... no
configure: error: The required function `tputs' was not found in any library.
The following libraries were tried (in order):
  libtinfo, libncurses, libterminfo, libtermcap, libcurses
```
Does it build if you just do `brew install --build-from-source emacs`?

**Comment by RandomDSdevel** [comment]
I'll try that next time I'm on my machine (I'm posting this from my 3DS.)

**Comment by RandomDSdevel** [comment]
Just tried `brew install -vd --build-from-source emacs --with-cocoa --with-d-bus --with-glib --with-gnutls --with-imagemagick --with-librsvg --with-mailutils` (that is, without `--with-x11`) and still got the same error.  I'll try removing `--with-cocoa` next and see if that works, but, if it doesn't, I'll go ahead and do a plain build-and-install without any of the options enabled.

**Comment by RandomDSdevel** [comment]
OK, `brew install -vd --build-from-source emacs --with-d-bus --with-glib --with-gnutls --with-imagemagick --with-librsvg --with-mailutils` (that is, without the `--with-cocoa` and `--with-x11` options) works perfectly well.  Now that I've done that, though, perhaps you could set up some formula build tests for those two options?  It would be nice if they could somehow be tied into `brew upgrade --all -vd --build-from-source` in such a way that running that command would notify me when those two options work again so I could `brew reinstall -vd --build-from-source emacs --with-cocoa --with-d-bus --with-glib --with-gnutls --with-imagemagick --with-librsvg --with-mailutils --with-x11`.

**Comment by dunn** [comment]
Could you post the `brew gist-logs emacs` for the `--with-cocoa` install attempt?

And you have XQuartz installed, correct?

**Comment by RandomDSdevel** [comment]
Yeah, I'll run it again when I have the time, and I do have XQuartz installed.  However, that latter installation is, per [issue #37803](https://github.com/Homebrew/homebrew/issues/37803), presumably part of the problem.

**Comment by RandomDSdevel** [comment]
OK, [here](https://gist.github.com/RandomDSdevel/2835249370a31bddcd2a) are the logs for a run of `brew reinstall -vd --build-from-source emacs --with-cocoa --with-d-bus --with-glib --with-gnutls --with-imagemagick --with-librsvg --with-mailutils`.

**Comment by dunn** [comment]
I've seen this error with Linuxbrew but I don't remember what the fix was, if I found one.

**Comment by dunn** [comment]
I didn't mean I'm giving up, just that I don't know how soon I'll solve this.

**Comment by RandomDSdevel** [comment]
@dunn:  OK, just let me know here when you've made some progress; I'm subscribed to this issue, so GitHub should ping me about updates to this thread on my notifications page.

**Comment by edipretoro** [comment]
I've got the exact same error a couple of days ago. As I couldn't find any solutions, I let things how they are until today. Then I continue to search for a solution. Here what I found :
* XCode was updated on my MBA during the last week;
* XCode Command Line Tools were previously installed but the update doesn't seem to include it, so you have to install it manually. Luckily, this can be done easily with the following command:  `xcode-select --install`
* After that, I reinstall emacs as I used to do previously and everything was perfectly fine. 

Hope this helps.

**Comment by dunn** [comment]
Thanks!

@RandomDSdevel, your last gist showed that you had the CLT installed. If you've upgraded to Xcode 7 since, does upgrading the CLT do anything?

**Comment by RandomDSdevel** [comment]
Huh, good question.  Let me try that and report back to you on that.

**Comment by RandomDSdevel** [comment]
Looks like I'm blocked by [Issue #44537:  'GTK+3 (v3.18.0) Failed to Upgrade/Build on OS X v10.11.0 'El Capitan''](https://github.com/Homebrew/homebrew/issues/44537) since the way I'm installing Emacs has a dependency on GTK+3.  I'll try again when that gets fixed.

**Comment by RandomDSdevel** [comment]
I don't see anything up there, so I'm going to add it.

**Comment by RandomDSdevel** [comment]
`brew reinstall -vd --build-from-source emacs --with-cocoa --with-d-bus --with-glib --with-gnutls --with-imagemagick --with-librsvg --with-mailutils --with-x11` still doesn't work.  I'll upload some more logs and investigate some more later.

**Comment by skuro** [comment]
I had a similar issue, and after checking the [detail logs](https://gist.github.com/anonymous/51d38f3beed70164efd6#file-config-log-L4527) (and thanks to a [useful comment](https://github.com/Homebrew/homebrew/issues/13261#issuecomment-6833964)) I pointed my finger to `librsvg`. 

The following were enough for me to fix the issue, maybe they'll be helpful for you as well:

```
brew rm -f librsvg
brew install librsvg
brew install -vd emacs
```

**Comment by RandomDSdevel** [comment]
`--with-cocoa` must work, then, but I'll probably still have trouble with `--with-x11`.

**Comment by tdsmith** [comment]
@dunn Is there anything you want to do here?

**Comment by dunn** [comment]
No, I'm afraid I never got to the bottom of this one.

**Comment by RandomDSdevel** [comment]
@apjanke, @dunn, @edipretoro, @skuro, @tdsmith:  Hmm, I may try doing this again without the `--with-x11` option now that [Homebrew Core issue #3531:  "emacs:  cleanup formula"](https://github.com/Homebrew/homebrew-core/pull/3531) has removed it.

**Comment by MikeMcQuaid** [comment]
We'll accept PRs for this but we're not actively working on it at this time.

**Comment by RandomDSdevel** [comment]
@MikeMcQuaid:  I'll put it on my to-do list, then.  I assume that I should base any new PR I might make against Homebrew Core instead of against Homebrew Legacy, though, right?

**Comment by MikeMcQuaid** [comment]
Yes, please.

**Comment by RandomDSdevel** [comment]
@MikeMcQuaid:  I'll do that, then.
